core/arch/arm/kernel/tee_ta_manager.c

```diff
@@ -408,7 +408,9 @@ static TEE_Result tee_ta_load_check_shdr(struct shdr *shdr)
 				SHDR_GET_SIG(shdr), shdr->sig_size);
 out:
 	crypto_ops.acipher.free_rsa_public_key(&key);
-	return res;
+	if (res != TEE_SUCCESS)
+		return TEE_ERROR_SECURITY;
+	return TEE_SUCCESS;
```

Comment:
There is still a possibility that the function will return return neither `TEE_SUCCESS` nor `TEE_ERROR_SECURITY`: it is `TEE_ERROR_NOT_SUPPORTED` line 392.